File: 69.SquareRoot.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0 or x == 1:
            return x
        n = 1
        prev_n = -1
        while prev_n != n:
            prev_n = n
            n = 0.5 * (n + (x / n))
        return int(n)

File: 69.SquareRoot.java
class Solution {
    public int mySqrt(int x) {
        long X = x;
        if (x == 0 || x == 1) return x;

        long low = 0, high = X;
        long ans = x;
        while (low <= high) {
            long mid = (low + high) / 2;
            if ((mid * mid) <= X) {
                ans = mid;
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return (int)ans;
    }
}
